src/Language/Haskell/Liquid/Bare/OfType.hs

```diff
@@ -100,9 +102,10 @@ rtypePredBinds = map uPVar . ty_preds . toRTypeRep
 ofBRType :: (PPrint r, Reftable r)
          => (SourcePos -> RTAlias RTyVar SpecType -> [BRType r] -> r -> BareM (RRType r))
          -> (r -> BareM r)
+         -> (Expr -> BareM Expr)
          -> BRType r
          -> BareM (RRType r)
-ofBRType appRTAlias resolveReft
+ofBRType appRTAlias resolveReft resolveExpr
```

Comment:
What I mean by "remove" it is that, if some change were made to Fixpoint
(still need to retrace back and figure out what it is exactly), we could
derive the location information in all cases from just the data being
`resolve`d.

I think it had to do with changing one of the `Symbol`s in `Expr` to
`LocSymbol`.
On Mar 8, 2015 11:21 AM, "Eric Seidel" <notifications@github.com> wrote:

> In src/Language/Haskell/Liquid/Bare/OfType.hs
> <https://github.com/ucsd-progsys/liquidhaskell/pull/356#discussion_r26007498>
> :
>
> >           -> BRType r
> >           -> BareM (RRType r)
> > -ofBRType appRTAlias resolveReft
> > +ofBRType appRTAlias resolveReft resolveExpr
>
> Ah quite right, I saw the Loc in the RApp case and I guess I stopped
> reading :)
>
> I suspect the Loc is used for error reporting, in which case we shouldn't
> just remove it outright. (There's another issue here somewhere that talks
> about adding Locs to all of the datatypes we use and making Located a
> type-class, but I can't find it in github's mobile site..)
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/ucsd-progsys/liquidhaskell/pull/356/files#r26007498>.
>